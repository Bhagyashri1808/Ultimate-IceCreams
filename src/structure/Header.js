import React from 'react';
import FocusLink from '../structure/FocusLink';
import iceCream from '../assets/img/ultimate-ice-cream.svg';

const Header = () => (
  <header>
    <h1>
      <img src={iceCream} alt="" />
      Ultimate Ice Cream
    </h1>
    <nav>
      <FocusLink to="/" activeClassName="active" exact>
        Menu
      </FocusLink>
      <FocusLink to="/ice-creams" activeClassName="active">
        Add Ice Cream
      </FocusLink>
    </nav>
  </header>
);

export default Header;
