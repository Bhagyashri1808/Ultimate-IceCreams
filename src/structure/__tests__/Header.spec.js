import React from 'react';
import { render } from 'react-testing-library';
import Header from '../Header';
import 'jest-dom/extend-expect';

describe('Header', () => {
  it('should render', () => {
    const { container } = render(<Header />);
    const img = container.firstChild.querySelector('header > h1 > img');
    expect(img).toHaveAttribute('alt', '');
    expect(img).toHaveAttribute('src', 'ultimate-ice-cream.svg');

    const allAnchors = container.firstChild.querySelectorAll('nav a');
    expect(allAnchors[0]).toHaveAttribute('href', '/');
    expect(allAnchors[1]).toHaveAttribute('href', '/ice-creams');

    const navContainer = container.firstChild.querySelector('nav');
    expect(navContainer).toMatchSnapshot();
  });
});
