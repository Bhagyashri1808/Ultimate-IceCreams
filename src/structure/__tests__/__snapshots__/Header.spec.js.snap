// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<nav>
  <a
    href="/"
  >
    Menu
    <span>
      {"focus":true}
    </span>
  </a>
  <a
    href="/ice-creams"
  >
    Add more frozen goodness
    <span>
      {"focus":true}
    </span>
  </a>
</nav>
`;
